test/unit/BondingManager.js

```diff
@@ -998,6 +998,109 @@ describe("BondingManager", () => {
         })
     })
 
+    describe("bondFor", () => {
+        let transcoder0
+        let transcoder1
+        let delegator1
+        let delegator2
+        const currentRound = 100
+
+        beforeEach(async () => {
+            transcoder0 = signers[0]
+            transcoder1 = signers[1]
+            delegator1 = signers[2]
+            delegator2 = signers[3]
+
+            await fixture.roundsManager.setMockBool(functionSig("currentRoundInitialized()"), true)
+            await fixture.roundsManager.setMockBool(functionSig("currentRoundLocked()"), false)
+            await fixture.roundsManager.setMockUint256(functionSig("currentRound()"), currentRound - 1)
+            await bondingManager.connect(transcoder0).bond(1000, transcoder0.address)
+            await bondingManager.connect(transcoder0).transcoder(5, 10)
+            await bondingManager.connect(transcoder1).bond(2000, transcoder1.address)
+            await bondingManager.connect(transcoder1).transcoder(5, 10)
+            await fixture.roundsManager.setMockUint256(functionSig("currentRound()"), currentRound)
+        })
+
+        describe("caller is unbonded", () => {
+            it("should set delegate for different address", async () => {
+                await bondingManager.connect(delegator1).bondForWithHint(
+                    1000,
+                    delegator2.address,
+                    transcoder0.address,
+                    ethers.constants.AddressZero,
+                    ethers.constants.AddressZero,
+                    ethers.constants.AddressZero,
+                    ethers.constants.AddressZero
+                )
+
+                assert.equal(
+                    (await bondingManager.getDelegator(delegator2.address))[2],
+                    transcoder0.address,
+                    "wrong delegateAddress"
+                )
+            })
+
+            it("should update delegate and bonded amount for a different address", async () => {
+                const startDelegatedAmount = (await bondingManager.getDelegator(transcoder0.address))[3]
+                await bondingManager.connect(delegator1).bondForWithHint(
+                    1000,
+                    delegator2.address,
+                    transcoder0.address,
+                    ethers.constants.AddressZero,
+                    ethers.constants.AddressZero,
+                    ethers.constants.AddressZero,
+                    ethers.constants.AddressZero
+                )
+                const endDelegatedAmount = (await bondingManager.getDelegator(transcoder0.address))[3]
+
+                assert.equal(endDelegatedAmount.sub(startDelegatedAmount), 1000, "wrong change in delegatedAmount")
```

Comment:
Thoughts on using `expect()` consistently going forward in new tests instead of `assert()`?